command/tls/ca/create/tls_ca_create.go

```diff
@@ -40,8 +48,8 @@ func (c *cmd) Run(args []string) int {
 		return 1
 	}
 
-	certFileName := fmt.Sprintf("%s-agent-ca.pem", c.prefix)
-	pkFileName := fmt.Sprintf("%s-agent-ca-key.pem", c.prefix)
+	certFileName := fmt.Sprintf("agent-%s-ca.pem", c.domain)
```

Comment:
The reason why I removed the consul in the beginning is that it gets ridiculous for certs with the consul domain: `consul-server-dc1-consul-0.pem`. Since was gone for certs, I removed it for the CA as well. I chose the order because this is also what the SAN looks like `server-dc1-consul-0.pem` becomes `server.dc1.consul`. 
I agree that having the domain in the first for the filename place reads a little nicer, but it then no longer has the same order as the  SAN.

Reply:
I'm not sure I understand.

`consul-example-com-agent-ca.pem`

seems more consistent with

`server-dc1-consul-0.pem`

than `agent-consul-example-com-ca.pem` to me?

Not a huge deal though, I'll leave it to you but maybe listing a full example of all the CA and cert filenames for both default and custom domain might help double check they aren't confusingly inconsistent anywhere?

Reply:
This is the current state:

| dc,domain  | ca | server | client | cli
| ---- | ---- | ---- | ---- | ---- |
| default, default | agent-consul-ca.pem  |  server-dc1-consul-0.pem |  client-dc1-consul-0.pem |  cli-dc1-consul-0.pem
| dc2, example.com | agent-example.com-ca.pem  | server-dc2-example.com-0.pem | client-dc2-example.com-0.pem | cli-dc2-example.com-0.pem

Your proposal:

| dc,domain  | ca | server | client | cli
| ---- | ---- | ---- | ---- | ---- |
| default, default | consul-consul-agent-ca.pe  |  server-dc1-consul-0.pem |  client-dc1-consul-0.pem |  cli-dc1-consul-0.pem
| dc2, example.com |consul-example-com-agent-ca.pe  | server-dc2-example.com-0.pem | client-dc2-example.com-0.pem | cli-dc2-example.com-0.pem

Reply:
Ah, I see. No my proposal was meant to be (I may have mis-stated it) "<domain>-agent-ca.pem"

So it would `consul-agent-ca.pem` by default (which reads nicely to me) but with `consul.example.com` as the domain `consul-example-com-agent-ca.pem`.

Like I said no big deal. The one thing that makes it worth nailing this down and merging soon though is that the Encryption guide @kaitlincarter-hc is working on in #5059 would really benefit from showing the same default file names that this tool creates!

Reply:
ok, I will change it so that the domain is first:

| dc,domain  | ca | server | client | cli
| ---- | ---- | ---- | ---- | ---- |
| default, default | consul-agent-ca.pem  |  consul-server-dc1-0.pem |  consul-client-dc1-0.pem |  consul-cli-dc1-0.pem
| dc2, example.com | example.com-agent-ca.pem  | example.com-server-dc2-0.pem | example.com-client-dc2-0.pem | example.com-cli-dc2-0.pem

Reply:
🎉 Let's merge this thing!